app/models/user.rb

```diff
@@ -10,6 +10,19 @@ class User < ApplicationRecord
   has_many :ingredients, through: :user_ingredients
   has_many :messages
 
+  has_many :requester_friendships, class_name: 'Friendship', foreign_key: 'requester_id'
+  has_many :requested_friendships, class_name: 'Friendship', foreign_key: 'requested_id'
+
   validates :username, presence: true, uniqueness: true, length: { minimum: 2 }
   validates :first_name, presence: true
+
+  def friends
+    friend_ids = requester_friendships.where(status: "accepted").pluck(:requested_id) +
+                 requested_friendships.where(status: "accepted").pluck(:requester_id)
+    User.where(id: friend_ids)
+  end
```

Comment:
Here, I go into both sets of friendships, whether our user (let's say Amie) has requested someone to be her friend or someone has sent her the invite. I then take out the friend's IDs from each set.
Then, I run line 22, which in SQL is something like SELECT * FROM users WHERE id... and goes through the array of friends_ids